core/src/main/java/software/amazon/awssdk/async/AsyncResponseHandler.java

```diff
@@ -46,6 +49,8 @@
      * will be treated as a failure of the response and the {@link #exceptionOccurred(Throwable)} callback will be invoked.
      * </p>
      *
+     * <p>This callback may never be called if the response has no content or if an error occurs.</p>
```

Comment:
Thanks, I'm a little confused. For non-streaming content, will this method be called?